crates/wasmparser/src/validator/component.rs

```diff
@@ -1149,22 +1158,21 @@ impl ComponentState {
                         | (ComponentEntityType::Component(_), ComponentEntityType::Component(_))
                         | (ComponentEntityType::Instance(_), ComponentEntityType::Instance(_))
                         | (ComponentEntityType::Func(_), ComponentEntityType::Func(_))
-                        | (ComponentEntityType::Value(_), ComponentEntityType::Value(_)) => {}
+                        | (ComponentEntityType::Value(_), ComponentEntityType::Value(_))
+                        | (ComponentEntityType::Type { .. }, ComponentEntityType::Type { .. }) => {}
                         _ => {
                             bail!(
                                 offset,
-                                "expected component instantiation argument \
-                                 `{name}` to be of type `{}`",
-                                expected.desc()
+                                "expected component instantiation argument `{name}` to be a {desc}",
+                                desc = expected.desc()
                             )
                         }
                     };
 
                     if !ComponentEntityType::internal_is_subtype_of(arg, types, expected, types) {
                         bail!(
                             offset,
-                            "{} type mismatch for component instantiation argument `{name}`",
-                            expected.desc(),
+                            "type mismatch for component instantiation argument `{name}`"
```

Comment:
FYI, the reason these two messages changed was that they didn't read so well when the description was "type".